File: solved/100_doors_php/java/Door.java
public class Door
{
	private boolean open = false;

	public Door()
	{
		this.open = false;
	}

	public Door(boolean open)
	{
		this.open = open;
	}

	public boolean isOpen()
	{
		return this.open;
	}

	public boolean isClose()
	{
		return !this.open;
	}

	public void open()
	{
		this.open = true;
	}

	public void close()
	{
		this.open = false;
	}

	public void flip()
	{
		if(this.isOpen())
		{
			this.close();
		}
		else
		{
			this.open();
		}
	}

	public String toString()
	{
		if(this.isOpen())
		{
			return new String("[ ] ");
		}
		else
		{
			return new String("[X] ");
		}
	}

}

File: solved/100_doors/php/pn/Door.php
<?php

  /**
  * @author PN @since 2014-08-13
  */
  class Door {

    const OPENED = '[ ]';
    const CLOSED = '[X]';

    /** @var boolean */
    private $isOpen;

    /**
    * @return Door
    * @author PN @since 2014-08-13
    */
    public function __construct($isOpen = FALSE) {
      if ( is_bool($isOpen) ) {
        $this->isOpen = $isOpen;
      } else {
        throw new \InvalidArgumentException();
      } // end if
    } // end method

    /**
    * @return void
    * @author PN @since 2014-08-13
    */
    public function open() {
      $this->isOpen = TRUE;
    } // end method

    /**
    * @return void
    * @author PN @since 2014-08-13
    */
    public function close() {
      $this->isOpen = FALSE;
    } // end method

    /**
    * @return boolean
    * @author PN @since 2014-08-13
    */
    public function isOpen() {
      return $this->isOpen;
    } // end method

    /**
    * @return string
    * @author PN @since 2014-08-13
    */
    public function __toString() {
      return ( $this->isOpen() ? self::OPENED : self::CLOSED );
    } // end method

  } // end class